compositor_pipeline/src/pipeline.rs

```diff
@@ -137,19 +126,26 @@ impl<Output: PipelineOutput> Pipeline<Output> {
     pub fn register_output(
         &self,
         output_id: OutputId,
-        output_opts: OutputOptions<Output>,
+        encoder_opts: EncoderOptions,
+        output_opts: OutputOptions,
     ) -> Result<(), RegisterOutputError> {
         if self.outputs.contains_key(&output_id) {
             return Err(RegisterOutputError::AlreadyRegistered(output_id));
         }
 
-        if output_opts.resolution.height % 2 != 0 || output_opts.resolution.width % 2 != 0 {
+        let EncoderOptions::H264(ref opts) = encoder_opts;
```

Comment:
I thought enums always required `else` branch even if there was only 1 variant. That's cool